gobblin-data-management/src/main/java/org/apache/gobblin/data/management/retention/dataset/MultiVersionCleanableDatasetBase.java

```diff
@@ -327,7 +327,9 @@ private boolean cleanDatasetVersions(List<T> versions, VersionSelectionPolicy<T>
     boolean isCleanSuccess = true;
     Collections.sort(versions, Collections.reverseOrder());
     Collection<T> deletableVersions = selectionPolicy.listSelectedVersions(versions);
-    cleanImpl(deletableVersions);
+    if (!deletableVersions.isEmpty()) {
```

Comment:
but then it will never print that log line.
can you distinguish the case when the caller has that iterator config set? i see there are `properties` coming into the constructor of this class.

Reply:
Thanks for pointing out, fixed it by passing the variable  `shouldLogDeletableVersion` via caller